crates/legacy/src/service.rs

```diff
@@ -1125,13 +1191,27 @@ pub async fn run_non_permissioned_standalone_builder_service<
                         .max_block_size
                 };
 
-                handle_received_txns(
+                let results = handle_received_txns(
                     &tx_sender,
-                    transactions,
+                    transactions.clone(),
                     TransactionSource::HotShot,
                     max_block_size,
                 )
                 .await;
+                let pairs: Vec<(<Types as NodeType>::Transaction, Result<_, _>)> = (0
+                    ..transactions.len())
+                    .map(|i| (transactions[i].clone(), results[i].clone()))
+                    .collect();
+                for (txn, res) in pairs {
```

Comment:
We don't set other transactions to Pending here

Reply:
Same here. The initialization status of all the transaction will be Pending, therefore we only set sequenced or rejected additionally.

Reply:
We don't initialize transactions to Pending at all when receiving from HotShot though